base_decorator.rb

```diff
@@ -0,0 +1,14 @@
+# Decorator inherits from nameable
+require_relative './nameable'
+
+class Decorator < Nameable
+  # constructor
+  def initialize(_nameable)
+    super()
+    @name = name
+  end
+
+  def correct_name
+    @nameable
```

Comment:
[REQUIRED] You are expected to call the correct_name method present in @nameable here.